fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -140,6 +155,106 @@ instance MonadDB (Transaction Server) Server where
 instance MonadRelayStore Server where
   getStoreVar = asks linkRelayStoreVar
 
+instance MonadPowerDNS Server where
+  set recType url zoneTxt contents ttl = do
+    logDebug $ mconcat
+      [ "Set DNS "
+      , displayShow recType
+      , " Record "
+      , displayShow url
+      , " to "
+      , displayShow contents
+      ]
+    clientEnv <- pdnsClientEnv
+    let
+      rrset =
+        RRSet { rrset_name = Text.append (textDisplay url) "."
+              , rrset_type = recType
+              , rrset_ttl = ttl
+              , rrset_changetype = Just Replace
+              , rrset_records = Just $ formatRecords contents
+              , rrset_comments = Nothing
+              }
+      rrsets = RRSets [rrset]
+    resp <- liftIO $ runClientM (PDNS.updateRecords "localhost" zoneTxt rrsets) clientEnv
+    case resp of
+      Left err -> do
+        logWarn @Text "PowerDNS.set failed"
+        return . Left $ Web.Error.toServerError err
+      Right good -> do
+        logDebug @Text "PowerDNS.set succeeded"
+        return $ Right good
+    where
+      formatRecords :: NonEmpty Text -> [Record]
+      formatRecords values =
+        fmap newRecord (toList values)
+
+      newRecord :: Text -> Record
+      newRecord content =
+        Record { record_content = format recType content, record_disabled = False }
+
+      format :: PowerDNS.API.Zones.RecordType -> Text -> Text
+      format TXT = UTF8.wrapIn "\""
+      format _   = identity
```

Comment:
Just the TXT records? 🤔 I'll read the rest, but when I see `format` in Haskell, I immediately wonder if there's a class we could use, like `ToJSON` or `Display`. Probably overkill for now, but conceptually we may later want a `class ToDNS a`.

Reply:
All the more reason to extract it!

...at some point when we're not running out of DNS space